src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
+      final UnitAttachment ua = UnitAttachment.get(airUnit.getType());
+      final int cost = ua.getCarrierCost();
+      if (cost != -1) {
```

Comment:
I think it is worth changing to > 0 still.
This check is already confusing, the horse is out of the barn and inconsistency won't make that confusion any worse. 

Having the reader recognize this pattern in the (hopefully just the) 2 or 3 places it exists is of small value. Every new reader, or reader that has lost context, will need to learn what this magic "-1" encoding means. That should stop most readers as magic int encoding is a pretty well known anti-pattern, causes confusing code and defects. 

The anti-pattern does reveal a smell, the code is pretty procedural. An "AircraftCarrier" object should be offered units, or a list of units, and that object should then be responsible for determining which planes can land. That can then be extended very cleanly to take into cases where we want highest defense value, or most planes.

```
final Collection<AircraftCarrier> carriers = Match.getMatches(units, .. UnitType.Carrier);
List<Unit> airUnitsToBeLanded = airUnits;
List<Unit> airThatCanLand = new ArrayList<>;
for( AircraftCarrier ac : carriers ) {
   List<Unit> airLandingOnAc = ac.selectPlanesToLand( airUnitsToBeLanded );
   airUnitsToBeLanded.removeAll(airLandingOnAc);
}
return airUnitsToBeLanded;
```

Then the extensions for policy:
```
   List<Unit> airLandingOnAc = ac.selectPlanesToLand( airUnitsToBeLanded, LandingPolicy.HIGHEST_DEFENSE );
   List<Unit> airLandingOnAc = ac.selectPlanesToLand( airUnitsToBeLanded, LandingPolicy.MAX_PLANES );
   List<Unit> airLandingOnAc = ac.selectPlanesToLand( airUnitsToBeLanded, LandingPolicy.MIN_PU_COST );
```